htdocs/core/lib/pdf.lib.php

```diff
@@ -400,9 +400,16 @@ function pdfBuildThirdpartyName($thirdparty, Translate $outputlangs, $includeali
 			}
 		}
 	} elseif ($thirdparty instanceof Contact) {
-		$socname = $thirdparty->socname;
-		if (($includealias || getDolGlobalInt('PDF_INCLUDE_ALIAS_IN_THIRDPARTY_NAME')) && !empty($thirdparty->name_alias)) {
-			// TODO PDF_INCLUDE_ALIAS_IN_THIRDPARTY_NAME not completely implemented
+		if ($thirdparty->socid > 0) {
+			$thirdparty->fetch_thirdparty();
```

Comment:
You need this fetch to get thirdparty->thirdparty->name_alias.
But we need it only it 
`if (($includealias || getDolGlobalInt('PDF_INCLUDE_ALIAS_IN_THIRDPARTY_NAME'))`
So to avoid useless fetch, can you move it after the if ?